test/hotspot/jtreg/serviceability/jvmti/RedefineClasses/RedefineRetransform/RedefineRetransform.java

```diff
@@ -0,0 +1,256 @@
+            throw new RuntimeException("Redefine error (ver = " + ver + ")");
+        }
+        // verity ClassFileLoadHook get expected class bytes
+        int hookVer = getClassBytesVersion(hookClassBytes);
+        if (hookVer != ver) {
+            throw new RuntimeException("CLFH got unexpected version: "  + hookVer
+                    + " (expected " + ver + ")");
+        }
+    }
+
+    static void retransform(int ver, int expectedVer) {
+        byte[] classBytes = getClassBytes(ver);
+        byte[] hookClassBytes = nRetransform(TestClass.class, classBytes);
+        int hookVer = getClassBytesVersion(hookClassBytes);
+        if (hookVer != expectedVer) {
+            throw new RuntimeException("CLFH got unexpected version: "  + hookVer
+                    + " (expected " + expectedVer + ")");
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        int testCase;
+        try {
+            testCase= Integer.valueOf(args[0]);
+        } catch (Exception ex) {
+            throw new RuntimeException("Single numeric argument expected", ex);
+        }
+        init();
+        switch (testCase) {
+        case 1:
+            test("Redefine-Retransform-Retransform", () -> {
+                redefine(1);
+                retransform(2, 1);
+                retransform(3, 1);
+            });
+            break;
+
+        case 2:
+            test("Redefine-Retransform-Redefine-Redefine", () -> {
+                redefine(1);
+                retransform(2, 1);
+                redefine(3);
+                redefine(4);
+            });
+            break;
+
+        case 3:
+            test("Redefine-Retransform-Redefine-Retransform", () -> {
+                redefine(1);
+                retransform(2, 1);    // sets cached class bytes
+                redefine(3);                    // resets cached class bytes
+                retransform(4, 3);
+            });
+            break;
+
+        case 4:
+            test("Retransform-Redefine-Retransform-Retransform", () -> {
+                retransform(1, 0);    // sets cached class bytes
+                redefine(2);                    // reset cached class bytes
+                retransform(3, 2);    // sets cached class bytes
+                retransform(4, 2);
+            });
+            break;
+
+        case 5:
+            test("Redefine-Retransform-Redefine-Retransform with CFLH", () -> {
+                redefine(1, 5);    // sets cached class bytes
+                retransform(2, 1);
+                redefine(3, 6);    // updates cached class bytes
+                retransform(4, 3);
+            });
+            break;
+
+        case 6:
+            test("Retransform-Redefine-Retransform-Retransform with CFLH", () -> {
+                retransform(1, 0);    // sets cached class bytes
+                redefine(2, 5);    // updates cached class bytes
+                retransform(3, 2);
+                retransform(4, 2);
```

Comment:
The comments for all test cases need to be aligned.
Also, it is better to comment all redefine/retransform cases.